openedx/core/djangoapps/lang_pref/middleware.py

```diff
@@ -51,25 +53,32 @@ def process_request(self, request):
     def process_response(self, request, response):
         # If the user is logged in, check for their language preference
         if getattr(request, 'user', None) and request.user.is_authenticated():
-            # Get the user's language preference
-            try:
-                user_pref = get_user_preference(request.user, LANGUAGE_KEY)
-            except (UserAPIRequestError, UserAPIInternalError):
-                # If we can't find the user preferences, then don't modify the cookie
-                pass
+            user_pref = None
+
+            anonymous_cookie_lang = getattr(request, '_anonymous_user_cookie_lang', None)
+            if anonymous_cookie_lang:
+                user_pref = anonymous_cookie_lang
+                set_user_preference(request.user, LANGUAGE_KEY, anonymous_cookie_lang)
+            else:
```

Comment:
Just writing the logic out here...

If the anonymous user cookie is set, it should supersede the user's account language preference. 

The anonymous user cookie should only be set when a user has made an explicit action (going directly to edx.org/es or using the language selector). In these cases, overriding the user's account preference is expected.